daemons/execd/remoted_schemas.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2023 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <ftw.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+#include <crm/cib.h>
+#include <crm/cib/cib_types.h>
+#include <crm/cib/internal.h>
+#include <crm/crm.h>
+#include <crm/msg_xml.h>
+#include <crm/common/mainloop.h>
+
+#include "pacemaker-execd.h"
+
+static int
+rm_files(const char *pathname, const struct stat *sbuf, int type, struct FTW *ftwb)
+{
+    /* Don't delete PCMK__REMOTE_SCHEMA_DIR . */
+    if (pcmk__str_eq(pathname, PCMK__REMOTE_SCHEMA_DIR, pcmk__str_none)) {
+        return 0;
+    }
+
+    if (remove(pathname) != 0) {
+        int rc = errno;
+        crm_err("Could not remove %s: %s", pathname, pcmk_rc_str(rc));
+        return -1;
+    }
+
+    return 0;
+}
+
+static void
+clean_up_extra_schema_files(void)
+{
+    struct stat sb;
+    int rc;
+
+    rc = stat(PCMK__REMOTE_SCHEMA_DIR, &sb);
+
+    if (rc == -1) {
+        if (errno == ENOENT) {
+            /* If the directory doesn't exist, try to make it first. */
+            if (mkdir(PCMK__REMOTE_SCHEMA_DIR, 0755) != 0) {
+                rc = errno;
+                crm_err("Could not create directory for schemas: %s",
+                        pcmk_rc_str(rc));
+            }
```

Comment:
presumably you want to start over if this succeeds? right now it would just return

Reply:
I'm not sure I quite follow - it doesn't help that I'm using the same message twice and github doesn't display a lot of context.  There's several cases to consider here:

* The directory doesn't exist, and mkdir succeeds - we don't need to do anything else here.
* The directory doesn't exist, and mkdir fails - I don't think there's anything we could possibly do.  I think everything else involving the directory will fail properly.  It would be better if it didn't even try, though.
* stat returned some other error code - ???.  This could be that a file exists with the same name already, or we don't have permission to check, or all sorts of things.  I'm currently catching the case that something with the same name already exists by simply not doing anything.  I don't know what to do with all the other possibilities.

Reply:
Ah, what I missed is that if we have to create the directory, it's already clean :)